[dogstatsd] enable dogstatsd6 on agent5

truthbk opened the pull request:
### What does this PR do?

This PR allows running dogstatsd6 on agent5. It will enable loading of the relevant ini-style (legacy) agent5 configuration to extract relevant configuration for the process.

### Motivation

We want to be able to run dogstatsd6 on agent5.

truthbk committed:
[dogstatsd] adding a legacy flag, configuration option.

truthbk committed:
[dogstatsd] support legacy ini configuration.

[dogstatsd] fix compilation issues.

[glide] adding go-ini and updating deps.

[dogstatsd] defer log flushing, no need to call explicitly.

[dogstatsd6] do not start if not enabled.

[config][test] add ini-file tests.

truthbk committed:
[config][test] adding ini file fixture.

truthbk committed:
[dogstatsd] allow legacy file logging.

truthbk committed:
[dsd6] make the expvar port for dsd6 configurable.

olivielpeau reviewed (changes requested):
Nice stuff! Looks good overall!

Added a few comments, more or less important, let me know what you think

truthbk committed:
[dogstatsd][config] map legacy config vars logically to config options.

truthbk committed:
[config] remove unused `path`

masci reviewed (changes requested):
Let's try to keep this codebase pure, see my comment for details.

truthbk committed:
[dogstatsd6] put socket cmd line option back in place.

truthbk committed:
[config][test] remove unused config option.

truthbk committed:
[dogstatsd6] remove legacy code.

[config] bind log_file to env variable.

[dogstatsd] removing unneeded go-ini.

[dogstatsd] re-adding socket flag after accidental removal.

[dogstatsd6] remove unneeded legacy-config parsing code.

truthbk committed:
[dogstatsd6][test] removing legacy tests - no longer needed.

truthbk committed:
[dsd6] bind environment variable for stats port configuration.

truthbk committed:
Merge branch 'master' into jaime/dogstatsd6on5
